### PR TITLE
fix(ticket): predefined field always loaded

### PR DESCRIPTION
**Pull request by Rom1-B**
The values of the predefined fields in the template were displayed instead of the values recorded in the ticket.

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | !24947

Possibly related to PR #11421

### CODE REVIEW

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Review by cedric-anne — COMMENTED** [review]
What should be done if a field is predefined, but user explicitely removed the value on creation ?

Maybe the condition should be `if ($this->isNewItem())`.

**Comment by cedric-anne** [comment]
> What should be done if a field is predefined, but user explicitely removed the value on creation ?
> 
> Maybe the condition should be `if ($this->isNewItem())`.

Or maybe, even, a combination of both conditions, i.e. `if ($this->isNewItem() && (!array_key_exists($predeffield, $this->fields) || empty($this->fields[$predeffield]))) {`, to handle form reload correctly.

**Review by trasher — APPROVED** [review]

File: src/CommonITILObject.php [hunk 1]
```diff
@@ -520,7 +520,12 @@ protected function setPredefinedFields(ITILTemplate $tt, array &$options, array
                         ) {
                             $options[$predeffield]           = $predefvalue;
                             $predefined_fields[$predeffield] = $predefvalue;
-                            $this->fields[$predeffield]      = $predefvalue;
+                            if (
+                                ($this->isNewItem())
+                                && (!array_key_exists($predeffield, $this->fields) || empty($this->fields[$predeffield]))
```
Comment (cconard96) [inline src/CommonITILObject.php:525]: Some fields have default non-empty values. Wouldn't this make them impossible to predefine?

Example,
internal time to own is a string with the text "NULL"
urgency and impact are integers with the value 3
priority is a string "3" (No idea why it is a different type)

Comment (cedric-anne) [reply src/CommonITILObject.php:525]: Hum, it was changed in #10886.

Comment (orthagh) [reply src/CommonITILObject.php:525]: nope

Comment (cedric-anne) [reply src/CommonITILObject.php]: Ok, so I guess, to restore the behaviour from GLPI 9.5, we should remove the `$this->getType() == "Ticket" ||` condition on line 495, and remove condition added by current PR..

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**